consensus-types/blocks/proto.go

```diff
@@ -188,7 +188,7 @@ func (b *BeaconBlockBody) Proto() (proto.Message, error) {
 
 func initSignedBlockFromProtoPhase0(pb *eth.SignedBeaconBlock) (*SignedBeaconBlock, error) {
 	if pb == nil {
-		return nil, errNilBlock
```

Comment:
I decided to relax this assertion because of tests, especially fuzz. Fuzz test randomizing can result in `Block` or `Block.Body` being `nil`, which results in errors and failed tests. I don't know if there is a way to overcome this. Additionally, returning an error means that all test setups, not only fuzz tests, need to set `Block` and `Block.Body` fields even if they don't need them. Furthermore, these functions are not exported, and so this change is not too risky.

Reply:
I mean even with an error, this shouldn't cause a fuzz test to fail. The fuzz test should be able to handle nil blocks gracefully. Is this what you are seeing ?

Reply:
The fuzzer generates nil values and then the fuzz test calls `NewSignedBeaconBlock` which returns an error. This makes the whole test fail.

Reply:
The job of the fuzzer is to find crashes though, having a nil block shouldnt be an issue. We should be gracefully handling it, can you link which fuzz test causes issues ?